### ISSUE TITLE
PowerMonitor exception on MacOS on sleep or screensaver

### ISSUE THREAD

**Issue by daytonturner** [jitsi/jitsi-meet-electron#219]
Hi folks,

Testing out Jitsi-Meet-Electron and its working great!  One thing I've found, on MacOS, intermittently when the screensaver activates or deactivates (not sure which), I come back to a dialog box saying:

```
Uncaught Exception:
Error: Object has been destroyed
  at PowerMonitor.<anonymous> (/Users/turnerd/git/jitsi-meet-electron/dist/mac/Jitsi Meet.app/Contents/Resources/app.asar/node_modules/jitsi-meet-electron-utils/powermonitor/main.js:22:29)
  at PowerMonitor.emit (events.js:208:15)
```

Which seems to just be that powermonitor received an event it didnt know what to do with, but I'm not entirely sure how to go about fixing the issue. 

Has anyone else run into this?

**Comment by Toraih** [comment]
One of my users just reported the exact same problem!

I downloaded and compiled the source-code on march 31th with the fixes from https://github.com/ctramnitz/jitsi-meet-electron . The macOS version for both developer- and client-computer is 10.14.6 (Mojave).

**The error seems to occur when waking up from sleep-mode (not screensaver) and it does not appear when energy-option: “Prevent Computer from sleeping when display is turned off” is set.**

There's also a community-post about this:
https://community.jitsi.org/t/powermonitor-exception-on-jitsi-meet-electron/28501

**Comment by saghul** [comment]
Can you folks test the latest beta release? I believe this was ssolved there already.

**Comment by rubendibattista** [comment]
I can confirm the same issue in the latest beta when I close the App.

**Comment by saghul** [comment]
Sorry, my bad, I thought we were using the latest version of the utils package, but we're not. This should fix it: https://github.com/jitsi/jitsi-meet-electron/pull/242

**Comment by saghul** [comment]
Can you please test this one? https://github.com/jitsi/jitsi-meet-electron/releases/tag/v2.0.0

**Comment by rubendibattista** [comment]
<img width="414" alt="image" src="https://user-images.githubusercontent.com/8077364/78801911-8546a200-79bd-11ea-9a0d-461121503ba5.png">

Same problem: I removed all the configuration settings in `Application Support`, then started the App. 

This happens only at the first start. All the following executions seem to not trigger this problem.

**Comment by saghul** [comment]
Thanks for the heads up, we'll take a look.

**Comment by valentin-nasta** [comment]
I can confirm the same:

![Screen Shot 2020-04-14 at 2 54 51 PM](https://user-images.githubusercontent.com/2648486/79227999-1061e500-7e61-11ea-8be9-de0135867213.png)


```
$ plutil -p '/Applications/Jitsi Meet.app/Contents/Info.plist' | grep CFBundleShortVersionString
  "CFBundleShortVersionString" => "2.0.0"
```

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
```

**Comment by gufertum** [comment]
same her using Version 2.0.2

```Uncaught Exception:
Error: Object has been destroyed
    at PowerMonitor.<anonymous> (/Applications/Jitsi Meet.app/Contents/Resources/app.asar/node_modules/jitsi-meet-electron-utils/powermonitor/main.js:22:29)
    at PowerMonitor.emit (events.js:210:5)`
```

```
ProductName:	Mac OS X
ProductVersion:	10.15.4
```

**Comment by maddie-wang** [comment]
Same!!!! BUMP :(

**Comment by lahirudx** [comment]
Same here. When trying to force close.

**Comment by ATQO** [comment]
Same here. I have tested this with the official Jitsi Meet desktop app on OSX and it has the same issue.

**Comment by rubendibattista** [comment]
I think that devs are aware of the problem, maybe let's stop replying in the thread and maybe post just an emoji (👍) to notify of you also having the same problem on the first message of the thread?

**Comment by saghul** [comment]
This was fixed in 2.2 or 2.3 IIRC.
